CMakeLists.txt

```diff
@@ -20,6 +20,13 @@ if (CMAKE_VERSION VERSION_GREATER_EQUAL "3.24.0")
   cmake_policy(SET CMP0135 NEW)
 endif()
 
+
+if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
+  set(SUGGEST_BUILD_SAMPLES ON)
```

Comment:
Could you use
```
SHERPA_ONNX_SUGGEST_BUILD_SAMPLES
```

Reply:
No, because I don't feel like the option should be force-overridden. The user should still be able to set it themself, and this way they are able to do that.

Reply:
Typically when you build a project as a subdirectory you don't want any binaries to be generated by it, just libraries.